packages/myst-cli/src/process/mdast.ts

```diff
@@ -273,14 +286,17 @@ export async function postProcessMdast(
     ...(extraLinkTransformers || []),
     new StaticFileTransformer(session, file), // Links static files and internally linked files
   ];
-  linksTransform(mdastPost.mdast, state.file as VFile, {
+  linksTransform(mdast, state.file as VFile, {
     transformers,
     selector: LINKS_SELECTOR,
   });
-  resolveReferencesTransform(mdastPost.mdast, state.file as VFile, { state });
-  embedDirective(session, mdastPost.mdast, mdastPost.dependencies, state);
+  resolveReferencesTransform(mdast, state.file as VFile, { state });
+  embedDirective(session, mdast, dependencies, state);
+  if (simplifyFigures) {
+    transformPlaceholderImages(mdast, { imageExtensions });
```

Comment:
This does need to happen in post-process because figure embed content need to be resolved alongside placeholder images.

Reply:
Add this as comment in code.